content/en/blog/2020/show-source-ip/index.md

```diff
@@ -0,0 +1,335 @@
+
+{{< tip >}}
+A receiver may be configured to support both version 1 and version 2 of the
+protocol. Identifying the protocol version is easy:
+
+- if the incoming byte count is 16 or more and the first 13 bytes match the protocol signature block `\x0D\x0A\x0D\x0A\x00\x0D\x0A\x51\x55\x49\x54\x0A\x02`, the protocol is version 2.
+
+- otherwise, if the incoming byte count is 8 or more, and the 5 first characters match the `US-ASCII` representation of "PROXY"(`\x50\x52\x4F\x58\x59`), then the protocol must be parsed as version 1.
+
+- otherwise the protocol is not covered by this specification and the connection must be dropped.
+{{< /tip >}}
+
+{{< image width="100%" link="./aws-proxy-protocol.png" caption="AWS NLB portal to enable proxy protocol" >}}
+
+## Two set-ups for 80 and 443 separately
+
+Before going through the following steps, the AWS env is assumed ready with proper VPC, IAM, and Kubernetes setup.
+
+### Step 1: Install Istio with AWS NLB
+
+Blog [Configuring Istio Ingress with AWS NLB](../../../blog/2018/aws-nlb/) provides detailed steps to set up AWS IAM roles and enable the usage of AWS NLB by Helm. The users can also use other automation tools e.g. Terraform to achieve the same goal. In this blog, below comes more complete configurations in order to enable proxy protocol and `X-Forwarded-For` at the same time.
+
+{{< text yaml >}}
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-proxy-protocol: "*"
+    service.beta.kubernetes.io/aws-load-balancer-type: "nlb"
+    proxy.istio.io/config: '{"gatewayTopology" : { "numTrustedProxies": 2 } }'
+  labels:
+    app: istio-ingressgateway
+    istio: ingressgateway
+    release: istio
+  name: istio-ingressgateway
+{{< /text >}}
+
+### Step 2:  Create proxy-protocol Envoy Filter
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+  name: proxy-protocol
+  namespace: istio-system
+spec:
+  workloadSelector:
+    labels:
+      istio: ingressgateway
+  configPatches:
+  - applyTo: LISTENER
+    patch:
+      operation: MERGE
+      value:
+        listener_filters:
+        - name: envoy.filters.listener.proxy_protocol
+        - name: envoy.filters.listener.tls_inspector
```

Comment:
Istio has already added this filter

Reply:
And also donot we need to match a specific port?

Reply:
Thanks, Zhihu. Based on out setup with Istio 1.7.3,  we need to add these filters. No port is needed here. Proxy protocol work with any layer 7 protocols even when encrypted).